src/components/Calculator.js

```diff
@@ -1,28 +1,27 @@
-import React from 'react';
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
 import Screen from './Screen';
 import Keys from './Keys';
 import calculate from '../logic/calculate';
 
-class Calculator extends React.Component {
-  constructor(props) {
-    super(props);
-    this.onClick = this.onClick.bind(this);
-    this.state = { total: null, next: null, operation: null };
-  }
+const Calculator = () => {
+  const [memory, setMemory] = useState({ total: null, next: null, operation: null });
 
-  onClick(props) {
-    const calc = calculate(this.state, props.value);
-    this.setState(() => ({ total: calc.total, next: calc.next, operation: calc.operation }));
-  }
+  const onClick = ({ value }) => {
+    const calc = calculate(memory, value);
+    setMemory(calc);
+  };
 
-  render() {
-    return (
-      <>
-        <Screen display={this.state} />
-        <Keys state={this.state} onClick={this.onClick} />
-      </>
-    );
-  }
-}
+  onClick.PropTypes = {
+    value: PropTypes.string.isRequired,
+  };
```

Comment:
[OPTIONAL] I think you might have misused PropTypes here. PropTypes are made to indicate react what are the type of the props of a component, but onClick is a function, not a component.

Reply:
Thanks! It was a mental typo. Fixed in the next commit.